### PR TITLE
feat(rulesets): add message traits array path to asyncapi headers schema type object rule

### PR DESCRIPTION
**Pull request by CalemRoelofsSB**
Recreation of #2402 to fix issues around commitlint

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

Currently, the `asyncapi-headers-schema-type-object` will not trigger when the `headers` object is defined explicitly in the `message.traits` array. 
For example:
```yaml
...
channels:
  'users/{userId}/loggedIn':
    publish:
      message:
        traits:
          - headers
              type: integer
```

The new path added in this PR will match on that array and perform the assertion on the `type` field of any object in that array that has a top level key of `headers`.

### CODE REVIEW

**Comment by CalemRoelofsSB** [comment]
ping @P0lip

**Review by P0lip — APPROVED** [review]

**Comment by P0lip** [comment]
Looks good to me, although I'd wait for an approval from our AsyncAPI colleagues as they're the most familiar with it :)

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by CalemRoelofsSB** [comment]
@P0lip @jonaslagoni thank you for the review folks! Unfortunately, I don't have permissions to merge this branch into `develop` so if one of you could oblige, that would be most appreciated :)

**Pull request merged**
